static/app/utils/useNavigate.tsx

```diff
@@ -0,0 +1,28 @@
+import {useContext} from 'react';
+
+import {RouteContext} from 'sentry/views/routeContext';
+
+type NavigateOptions = {
+  replace?: boolean;
+  state?: any;
+};
+
+export function useNavigate() {
+  const route = useContext(RouteContext);
+  if (route === null) {
+    throw new Error('useNavigate called outside of routes provider');
+  }
+  const navigator = route?.router;
+
+  const navigate = (to: string | number, options: NavigateOptions = {}) => {
```

Comment:
This should use useCallback to avoid unstable fn references causing rerenders down the tree